app/javascript/hooks/useTranslate.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * This hook is for using an older version of Google Translate
+ * that requires work to ensure the timing of script loading and event
+ * dispatching is correct. Essentially, we inject a Google Translate script that adds
+ * a <select> element to the dom. That <select> element is a dropdown of different languages
+ * that we can machine translate the page with. We hide that <select> with css.
+ * Once that <select> element is rendered (even though hidden), we use the language
+ * set in the url to dispatch an event setting that <select>'s <option>
+ * to the corresponding language. This gives an experience to the end user that
+ * the page automatically got translated.
+ */
+
+import { useEffect } from "react"
+import useLanguageStringInUrl from "./useLanguageStringInUrl"
+import usePollElementRender from "./usePollElementRender"
+import useScript from "./useScript"
+const tWindow = window as any
+const languageMap = {
+  zh: "zh-TW",
+}
+
+const initGoogleTranslate = () => {
+  // eslint-disable-next-line no-new
+  new tWindow.google.translate.TranslateElement(
+    {
+      pageLanguage: "en",
+      includedLanguages: "en,es,tl,zh-TW",
+      layout: tWindow.google.translate.TranslateElement.InlineLayout.HORIZONTAL,
+      autoDisplay: false,
+      multilanguagePage: true,
+    },
+    "google_translate_element"
+  )
+}
+
+tWindow.initGoogleTranslate = initGoogleTranslate
+
+const useTranslate = () => {
+  const languageInRoute = useLanguageStringInUrl()
+  useScript("//translate.google.com/translate_a/element.js?cb=initGoogleTranslate")
+  /*
+   * It seems the <select> and the <options> get added at different times.
+   * We want to target the <select> for dispatching the change event, but we have to wait
+   * until the <options> appear to dispatch the event.
+   */
+  const googleTranslateDropdownElHasBeenRendered = usePollElementRender(
+    "select.goog-te-combo option"
+  )
+
+  useEffect(() => {
+    if (languageInRoute && googleTranslateDropdownElHasBeenRendered) {
+      const selectInDom = document.querySelector("select.goog-te-combo")
+      selectInDom.value = languageMap[languageInRoute] || languageInRoute
+      const ev = new Event("change", { bubbles: true })
+      selectInDom.dispatchEvent(ev)
+    }
+  }, [languageInRoute, googleTranslateDropdownElHasBeenRendered])
+}
+
+export default useTranslate
```

Comment:
[an article](https://kentcdodds.com/blog/how-to-test-custom-react-hooks) on testing custom hooks.  not sure if it's relevant here. if we're unable to add any test coverage in this pr, then let's make sure to file an e2e ticket

Reply:
sounds good! I like that approach